runtime/server/canvases.go

```diff
@@ -143,6 +143,32 @@ func (s *Server) ResolveCanvas(ctx context.Context, req *runtimev1.ResolveCanvas
 		metricsViews[mvName] = mv
 	}
 
+	// Validate all metrics views have consistent first_day_of_week or first_month_of_year
+	// This ensures that time-based aggregations across different metrics views are consistent
+	if len(metricsViews) > 0 {
```

Comment:
The validation needs to go in `runtime/reconcilers/canvas.go` so that it runs as part of reconciliation/validation, not on every load.